obspy/io/nordic/core.py

```diff
@@ -0,0 +1,1379 @@
+                str(evtime.minute).rjust(2) + ' ' +
+                str(float(evtime.second)).rjust(4) + ' ' +
+                evtype + '1'.rjust(58) + '\n'))
+    # Write line 2 of s-file
+    f.write(str(' ACTION:ARG ' + str(datetime.datetime.now().year)[2:4] + '-' +
+                str(datetime.datetime.now().month).zfill(2) + '-' +
+                str(datetime.datetime.now().day).zfill(2) + ' ' +
+                str(datetime.datetime.now().hour).zfill(2) + ':' +
+                str(datetime.datetime.now().minute).zfill(2) + ' OP:' +
+                userid.ljust(4) + ' STATUS:' + 'ID:'.rjust(18) +
+                str(evtime.year) +
+                str(evtime.month).zfill(2) +
+                str(evtime.day).zfill(2) +
+                str(evtime.hour).zfill(2) +
+                str(evtime.minute).zfill(2) +
+                str(evtime.second).zfill(2) +
+                'I'.rjust(6) + '\n'))
+    # Write line 3 of s-file
+    write_wavfile = wavefile.split(os.sep)[-1]
+    f.write(str(' ' + write_wavfile + '6'.rjust(79 - len(write_wavfile)) +
+                '\n'))
+    # Write final line of s-file
+    f.write(str(' STAT SP IPHASW D HRMM SECON CODA AMPLIT PERI AZIMU' +
+                ' VELO AIN AR TRES W  DIS CAZ7\n'))
+    f.close()
+    return sfile
+
+
+def write_select(catalog, filename='select.out', userid='OBSP', evtype='L',
+                 wavefiles=None):
+    """Function to write a catalog to a select file in nordic format.
+
+    :type catalog: obspy.core.event.Catalog
+    :param catalog: A catalog of obspy events
+    :type filename: str
+    :param filename: Path to write to
+    :type userid: str
+    :param userid: Up to 4 character user ID
+    :type evtype: str
+    :param evtype: Single character string to describe the event, either L, R \
+        or D.
+    :type wavefiles: list
+    :param wavefiles: Waveforms to associate the events with, must be ordered \
+        in the same way as the events in the catalog.
+    """
+    if not wavefiles:
+        wavefiles = ['DUMMY' for i in range(len(catalog))]
+    with open(filename, 'w') as fout:
+        for event, wavfile in zip(catalog, wavefiles):
+            sfile = write_nordic(event=event, userid=userid, evtype=evtype,
+                                 wavefiles=wavfile)
+            with open(sfile, 'r') as f:
+                for line in f:
+                    fout.write(line)
+            fout.write('\n')
+            os.remove(sfile)
+
+
+def write_nordic(event, filename=None, userid='OBSP', evtype='L', outdir='.',
+                 wavefiles='DUMMY', explosion=False,
+                 overwrite=False):
+    """
+    Write an obspy.event to a nordic formatted s-file.
+
+    :type event: obspy.core.event.Event
+    :param event: A single obspy event
+    :type filename: str
+    :param filename: Filname to write to, can be None, and filename will be \
+        generated from the origin time in seisan format.
```

Comment:
Please require users to just pass a filename. This might result in hard to debug errors if this generated random file names.